infrastructure/ArmTemplates/KeyVaultDeploy.json

```diff
@@ -26,38 +26,46 @@
-        }
       }
-
     },
     "variables": {
       "tenantId": "[subscription().tenantId]"
      },
     "resources": [
         {
-            "type": "Microsoft.KeyVault/vaults",
-            "name": "[parameters('KeyVaultName')]",
-            "apiVersion": "2015-06-01",
-            "location": "[parameters('KeyVaultLocation')]",
+            "type": "Microsoft.KeyVault/vaults/accessPolicies",
+            "name": "[concat(parameters('keyVaultName'), '/add')]",
+            "apiVersion": "2019-09-01",
+            "properties": {
+                "copy" : [
+                    {
+                        "name": "accessPolicies",
+                        "count": "[length(parameters('sites'))]",
+                        "input": {
+                            "tenantId": "[variables('tenantId')]",
+                            "objectId": "[reference(concat(resourceId(parameters('subscriptionId'), parameters('ResourceGroup'), 'Microsoft.Web/sites', parameters('sites')[copyIndex('accessPolicies')].siteName),'/slots/staging/providers/Microsoft.ManagedIdentity/Identities/default'), '2015-08-31-PREVIEW').principalId]",
+                            "permissions": {
+                                "secrets": [
+                                    "get","list"
+                                ]
+                            } 
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "type": "Microsoft.KeyVault/vaults/accessPolicies",
+            "name": "[concat(parameters('keyVaultName'), '/add')]",
+            "apiVersion": "2019-09-01",
             "properties": {
-                "sku": {
-                    "family": "A",
-                    "name": "Standard"
-                },
-                "tenantId": "[variables('tenantId')]",
-                "enabledForDeployment": false,
                 "copy" : [
                     {
                         "name": "accessPolicies",
-                        "count": "[mul(length(parameters('sites')),2)]",
+                        "count": "[length(parameters('sites'))]",
                         "input": {
                             "tenantId": "[variables('tenantId')]",
-                            "objectId": "[reference(concat(resourceId(parameters('subscriptionId'), parameters('ResourceGroup'), 'Microsoft.Web/sites', parameters('sites')[div(copyIndex('accessPolicies'),2)].siteName),if(equals(mod(copyIndex('accessPolicies'),2),1),'/providers/Microsoft.ManagedIdentity/Identities/default','/slots/staging/providers/Microsoft.ManagedIdentity/Identities/default')), '2015-08-31-PREVIEW').principalId]",
+                            "objectId": "[reference(concat(resourceId(parameters('subscriptionId'), parameters('ResourceGroup'), 'Microsoft.Web/sites', parameters('sites')[copyIndex('accessPolicies')].siteName),'/providers/Microsoft.ManagedIdentity/Identities/default'), '2015-08-31-PREVIEW').principalId]",
```

Comment:
Craig, how would this work if I pass in array of sites each belonging to different resource group?

Reply:
It can be different according to region. Example EUAP runtimehosts belong a specific resource group, and am2 in other